okhttp/src/main/java/com/squareup/okhttp/FormEncodingBuilder.java

```diff
@@ -45,12 +45,32 @@ public FormEncodingBuilder add(String name, String value) {
   }
 
   public RequestBody build() {
-    if (content.length() == 0) {
+    if (content.size() == 0) {
       throw new IllegalStateException("Form encoded body must have at least one part.");
     }
+    return new FormEncodingRequestBody(content.snapshot());
```

Comment:
I figured `ByteString` would be optimal because if it's `Segment`-backed (as it is here) we could potentially pull those immutable segments back out when writing.